File: app/src/main/java/com/example/neon/deardiary/diaryedit/DiaryEditContract.java
package com.example.neon.deardiary.diaryedit;

import com.example.neon.deardiary.base.BasePresenter;
import com.example.neon.deardiary.base.BaseView;
import com.example.neon.deardiary.data.Diary;


interface DiaryEditContract {
    interface View extends BaseView<Presenter> {
        void onDiaryUpdated();

        void onDiaryUpdateFailed();
    }

    interface Presenter extends BasePresenter {
        void saveDiary(Diary diary);

        int getValidDiaryCount();

    }


}

File: app/src/main/java/com/example/neon/deardiary/diaryedit/DiaryEditContract.kt
package com.example.neon.deardiary.diaryedit

import com.example.neon.deardiary.base.BaseView
import com.example.neon.deardiary.data.Diary


internal interface DiaryEditContract {
    interface View : BaseView<Presenter> {
        fun onDiaryUpdated()

        fun onDiaryUpdateFailed()
    }

    interface Presenter {
        fun saveDiary(diary: Diary)

        fun validDiaryCount(): Int
    }


}
